funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/result_queue_publisher.py

```diff
@@ -0,0 +1,82 @@
+import logging
+
+import pika
+
+logger = logging.getLogger(__name__)
+
+
+class ResultQueuePublisher:
+    """ResultPublisher publishes results to a topic exchange_name, with
+    the {endpoint_id}.results as a routing key.
+    """
+
+    def __init__(
+        self,
+        endpoint_id: str,
+        pika_conn_params: pika.connection.Parameters,
+    ):
+        """
+        Parameters
+        ----------
+        endpoint_uuid: str
+            Endpoint UUID string used to identify the endpoint
+        pika_conn_params: pika.connection.Parameters
+            Pika connection parameters to connect to RabbitMQ
+        """
+        self.endpoint_id = endpoint_id
+        self.params = pika_conn_params
+        self._channel = None
+        self._connection = None
+
+        self.exchange_name = "results"
+        self.exchange_type = "topic"
+        self.queue_name = "results"
```

Comment:
In the comment about these, you said they were going to become class attrs, but these are instance attrs. I don't know that it really matters, but I was more in favor of class attrs because they can be overridden by subclassing if that's ever actually desired. And otherwise they are declared as part of the class body rather than buried in `__init__`.